executor/src/witgen/jit/symbolic_expression.rs

```diff
@@ -89,6 +111,18 @@ impl<T: FieldElement, S> SymbolicExpression<T, S> {
     }
 }
 
+impl<T: FieldElement, S: Ord> SymbolicExpression<T, S> {
+    pub fn referenced_symbols(&self) -> BTreeSet<&S> {
+        match self {
+            SymbolicExpression::Symbol(s, _) => iter::once(s).collect(),
+            _ => self
+                .children()
```

Comment:
I think it's cleaner using `all_children()` instead of a recursive call.

Reply:
I tried for a while but for some reason, `SymbolicExpression` does not implement `AllChildren`, despite the blanket implementation:
```
    = note: the following trait bounds were not satisfied:
            `&SymbolicExpression<T, S>: Children<_>`
            which is required by `&SymbolicExpression<T, S>: AllChildren<_>`
    = help: items from traits can only be used if the trait is implemented and in scope
    = note: the following trait defines an item `all_children`, perhaps you need to implement it:
            candidate #1: `AllChildren`
```

And when I implement it manually, I get the error that there is a conflicting implementation...

Don't really know what I'm doing wrong :/